packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -249,393 +314,657 @@ describe("Decentraland", () => {
-            `${assetId}`,
-            `${priceInWei}`,
-            `${fingerprint}`
-          );
+        beforeEach(
+          "onboarding - approving ephemeral account to spend Safe's MANA",
+          done => {
+            (async () => {
+              // Global hooks don't run between same level hooks
+              await mineBlocks(provider, 1);
+
+              // Gnosis Safe should approve ephemeral account to spend its MANA Tokens
+              // TODO: ephemeralWallet should broadcast this action
+              const contractAddress = mana.getAddress();
+              const contractABI = mana.getABI();
+              const functionName = "approve";
+              const spender = ephemeralAddress;
+              const argsArray = [spender, manaAmountForShopping];
+              gnosisSafe.setSigners([gnosisSafeOwner]);
+              gnosisSafe.setWallet(gnosisSafeOwner);
+              const action = gnosisSafe.customContractAction(
+                contractAddress,
+                contractABI,
+                functionName,
+                argsArray
+              );
+
+              const confirmationListener = async message => {
+                const owner = GNOSIS_SAFE_ADDRESS;
+                const ephemeralAllowance = await mana.allowance(owner, spender);
+                expect(`${ephemeralAllowance}`).to.equal(
+                  `${manaAmountForShopping}`
+                );
+                done();
+              };
+
+              const errorListener = error => {
+                action.off("error");
+                done(error);
+              };
+
+              action.once("confirmation", confirmationListener);
+              action.on("error", errorListener);
+
+              action.send();
+            })();
+          }
+        );
 
-          await executeOrderAction.send();
-          await executeOrderAction.waitForOneConfirmation();
+        beforeEach(
+          "onboarding - approving Marketplace to spend ephemeral account's MANA",
+          done => {
+            (async () => {
+              // Global hooks don't run between same level hooks
+              await mineBlocks(provider, 1);
+
+              // Ephemeral account approves Marketplace to spend their mana tokens (actually tokens from Gnosis Safe).
+              mana.setWallet(ephemeralWallet);
+              const action = mana.approve(
+                MARKETPLACE_ADDRESS,
+                manaAmountForShopping
+              );
+
+              const confirmationListener = async message => {
+                const marketplaceAllowance = await mana.allowance(
+                  ephemeralAddress,
+                  MARKETPLACE_ADDRESS
+                );
+                expect(`${marketplaceAllowance}`).to.equal(
+                  `${manaAmountForShopping}`
+                );
+                done();
+              };
+
+              const errorListener = error => {
+                action.off("error");
+                done(error);
```

Comment:
Isn't this the test `onboarding - approving Marketplace to spend ephemeral account's MANA` where for now we want to assert that it won't work? So should we be doing an expectation here that the error listener is triggered?

Reply:
Oh never mind, I see that that is done below - we're in the beforeEach here, and we're using done() syntax there. Got it.

Reply:
Could I resolve this?

Reply:
Yep, I'm resolving it now